### PR TITLE
feat: add error when labelMap is no match

### PR DESCRIPTION
**Pull request by holanlan**
新增识别category与labelMap不匹配的逻辑，防止花费大量时间对模型进行训练后才发现报错

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -111,6 +111,9 @@ const getValidPair = async (dataPath: string, labelMap: Record<string, number>)
     if (fs.existsSync(path.join(dataPath, imageName))) {
       cocoJson.images.push(imageItem);
       imageData.annotation.object.forEach((object: any) => {
+        if (typeof labelMap[object.name[0]] === 'undefined') {
+          throw `category: ${object.name[0]} is not match with train/*.xml (from ${fileName})`
```
Comment (yorkie) [inline src/index.ts]: 这里应该 throw 一个 TypeError? @rickycao-qy 帮忙看看。

**Review by yorkie — CHANGES_REQUESTED** [review]

File: src/index.ts [hunk 2]
```diff
@@ -223,7 +226,7 @@ const cocoDataAccess: DataAccessType = async (args: ArgsType): Promise<CocoDatas
   if (testPair.length > 0) {
     result.testLoader = testLoader;
   }
-
+
```
Comment (rickycao-qy) [inline src/index.ts]: May I ask you to remove these spaces here

File: src/index.ts (hunk 1 above)
Comment (rickycao-qy) [reply src/index.ts]: Yes, shall we construct a type error here, for instance, testNotMatchError?

File: src/index.ts (hunk 2 above)
Comment (holanlan) [reply src/index.ts]: ok,I have removed them, please check it.

File: src/index.ts (hunk 1 above)
Comment (yorkie) [reply src/index.ts]: Ping @holanlan, this needs an update, too.

Comment (holanlan) [reply src/index.ts]: ok, I used `TypeError` here.

**Review by yorkie — APPROVED** [review]

**Comment by holanlan** [comment]
I found another error after the training is finished. It seems that because the categorys of the validation/*.xml  is less than train/*.xml.

Should we check in this plugin?

**Review by rickycao-qy — APPROVED** [review]

**Comment by rickycao-qy** [comment]
> I found another error after the training is finished. It seems that because the categorys of the validation/_.xml is less than train/_.xml.
> 
> Should we check in this plugin?

If all categories in validation are included in train, that should be fine. The categories in validation are allowed to be less than train

**Comment by rickycao-qy** [comment]
> I found another error after the training is finished. It seems that because the categorys of the validation/_.xml is less than train/_.xml.
> 
> Should we check in this plugin?

I understand this issue has no relation with current PR. Would you mind raising another issue in current repo and provide sample data to reproduce this problem

**Comment by holanlan** [comment]
> > I found another error after the training is finished. It seems that because the categorys of the validation/_.xml is less than train/_.xml.
> > Should we check in this plugin?
> 
> I understand this issue has no relation with current PR. Would you mind raising another issue in current repo and provide sample data to reproduce this problem

ok, I will raising another issue when I reproduce this problem.

**Pull request merged**